src/esm_runscripts/echam.py

```diff
@@ -0,0 +1,176 @@
+        A list of mvstream tags found in the namelist.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the namelist file specified by the path cannot be found.
+    TypeError
+        If the provided namelist cannot be converted to an f90nml.namelist.Namelist
+        object.
+
+    Examples
+    --------
+    Assuming you have a namelist file at ``/tmp/example/namelist.echam`` with the
+    following contents:
+
+    .. code-block:: fortran
+
+    ! This is the "no output" variant of namelist.echam. It contains absolutely no
+    ! output, and can be used as a template for testing.
+    !
+    ! Extended by mvstreamctl namelist block, as an example for the mvstream tags.
+    !
+    ! P. Gierz
+    ! Alfred Wegener Institute, Helmholtz Centre for Polar and Marine Research
+    ! July 2021
+    !
+    ! P. Gierz
+    ! October 2024
+
+    &runctl
+        dt_start = 2285, 12, 31, 23, 52, 30
+        dt_stop = 6699, 12, 31, 23, 52, 30
+        putrerun = 12, 'months', 'last', 0
+        lfractional_mask = .false.
+        lresume = .true.
+        out_datapath = './'
+        out_expname = 'E280'
+        rerun_filetype = 4
+        delta_time = 450
+        putdata = 1, 'months', 'last', 0
+        nproma = 8
+        lcouple = .true.
+        getocean = 1, 'days', 'last', 0
+        putocean = 1, 'days', 'last', 0
+        lcouple_co2 = .true.
+        default_output = .false.
+    /
+
+    &parctl
+        nproca = 24
+        nprocb = 24
+    /
+
+    &submodelctl
+        lmethox = .true.
+    /
+
+    &submdiagctl
+        vphysc_lpost = .false.
+    /
+
+    &radctl
+        iaero = 3
+        io3 = 4
+        isolrad = 6
+        ich4 = 3
+        in2o = 3
+        co2vmr = 284.3169860840e-06
+        ch4vmr = 808.2490234375e-09
+        n2ovmr = 273.0210571289e-09
+        yr_perp = 1850
+    /
+    &mvstreamctl
+        filetag = 'paul_custom'
+        source = 'g3b'
+        variables = 'temp2:mean>temp2=167'
+        interval = 1, 'months', 'last', 0
+    /
+
+    The following code will extract the mvstream tags from the namelist:
+
+    >>> namelist_path = "/tmp/example/namelist.echam"
+    >>> tags = _get_mvstream_tags_from_namelist(namelist_path)
+    >>> print(tags)
+    ['paul_custom']
+    """
+    mvstream_tags = []
+    if not isinstance(namelist, f90nml.namelist.Namelist):
+        try:
+            namelist = f90nml.read(namelist)
+        except FileNotFoundError:
+            logger.error(f"Namelist specified by {namelist} could not be found")
+        except TypeError as e:
+            logger.error(
+                f"Could not convert {namelist} to f90nml.namelist.Namelist object."
+            )
+            raise e
+
+    for chapter, contents in namelist.items():
+        if chapter == "mvstreamctl":
+            tag = contents.get("filetag")
+            if tag is not None:
+                mvstream_tags.append(tag)
+    return mvstream_tags
+
+
+def append_namelist_dependent_sources(config):
```

Comment:
Shouldn't this happen for `echam` but also `jsbach`? Doesn't `jsbach` also have its own streams?

Reply:
This is a little bit convoluted: jsbach and echam are controlled by the same namelist, so I would need some way to filter out which tags go to which model. In principle this would just be defined by which model config they come from.

Reply:
See https://github.com/esm-tools/esm_tools/pull/1240/commits/509ac21d102e41dcaaa1e9cbe95dca73834cceb7.

Reply:
>  jsbach and echam are controlled by the same namelist

Do you mean that they can be controlled from the same namelist? If that is the case, then I understand the rest of your reply and the new `jsbach.py`

Reply:
Yes, the output generated from `jsbach` is defined in `mvstreamctl` chapters of `namelist.echam` (since JSBACH technically speaking is just a submodule inside of ECHAM), and does not have any output control in it's own namelist.